### PR TITLE
Upgrade to Cliquet 2

### PR DESCRIPTION
**Pull request by leplatrem**
I wanted to compare load tests performances now that *Cliquet* has Pypy support. So I started to port *Reading List*.

→ Good exercise, I found [a bug](https://github.com/mozilla-services/cliquet/issues/348) in Cliquet :)

* [x] Upgrade deps
* [x] Update tests payloads and responses
* [x] Clean-up code now available in Cliquet
* [x] Find an elegant fix for https://github.com/mozilla-services/cliquet/issues/348
* [x] Fix tests

### CODE REVIEW

File: readinglist/tests/test_views_article.py [hunk 1]
```diff
@@ -150,97 +152,67 @@ def test_read_position_is_reset_when_unread_is_set_to_true(self):
...
+        data = {
             'unread': False,
             'marked_read_by': 'Android',
             'marked_read_on': 543210}
         resp = self.app.patch_json(self.url,
-                                   body,
+                                   {'data': data},
                                    headers=self.headers)
-        self.assertEqual(resp.json['last_modified'],
+        print resp.json['data']
```
Comment (Natim) [inline readinglist/tests/test_views_article.py]: print?

**Comment by Natim** [comment]
We should update the major version readinglist since this break the api (at least the new {"data": {}} formalism)

**Comment by ametaireau** [comment]
That's actually quite interesting to find out what are the breaking changes we did with the new versions of cliquet, and it's quite managable!

Also, this would break the clients because we're now returning the data attribute and using Etags rather than LMS.

**Comment by leplatrem** [comment]
Tests now pass locally, will have to merge and release with this fix though: https://github.com/mozilla-services/cliquet/issues/371

File: readinglist/__init__.py [hunk 2]
```diff
@@ -31,12 +37,8 @@ def main(global_config, **settings):
 
     patch_gevent(settings)
 
-    cliquet.initialize(config, version=__version__)
-
-    # Force default pagination (if empty, None or 0)
-    paginate_by = config.registry.settings.get('cliquet.paginate_by')
-    if not paginate_by:
-        config.registry.settings['cliquet.paginate_by'] = 100
+    cliquet.initialize(config, version=__version__,
```
Comment (Natim) [inline readinglist/__init__.py]: :+1:

**Comment by leplatrem** [comment]
> We should update the major readinglist version
Let's do that in a dedicated PR for release

**Pull request merged**